lib/index.js

```diff
@@ -170,6 +170,136 @@ class VideoAsset extends BaseAsset {
 
 exports.VideoAsset = VideoAsset;
 
+function _ensure_cheerio(value) {
+    if (typeof value === 'string')
+        return cheerio.load(value);
+    return value;
+}
+
+class BlogArticle extends BaseAsset {
+    constructor() {
+        super();
+        this._object_type = "ArticleObject";
+        this._content_type = "text/html";
+    }
+
+    set_author(value) { this._author = value; }
+    set_main_image(value) { this._main_image = value; }
+    set_main_image_caption(value) { this._main_image_caption = value; }
+    set_body(value) { this._body = _ensure_cheerio(value); }
+    set_read_more_text(value) { this._read_more_text = value; }
+    set_tags (value) { this._tags = value; }
+    set_as_static_page() { this._ekn_tags.push(EKN_STATIC_TAG); }
+
+    set_date_published(value) {
+        this._date_published = value;
+        if (!(value instanceof Date))
+            this._date_published = new Date(value);
+    }
+
+    to_metadata() {
+        const metadata = super.to_metadata();
+        Object.assign(metadata, {
+            'document': this._document.html(),
+            'author': this._author,
+            'published': this._date_published.toISOString(),
+        });
+        return metadata;
+    }
+
+    _process() {
+        return Promise.resolve().then(() => {
+            const $elem = this._document;
+
+            $elem('img, video, audio').each(function() {
+                const $asset = cheerio(this);
+
+                const asset_id = $asset.attr('data-libingester-asset-id');
+                if (asset_id) {
+                    $asset.attr('data-soma-job-id', asset_id);
+                    $asset.attr('data-libingester-asset-id', null);
+                }
+
+                if (!$asset.attr('data-soma-job-id')) {
+                    throw new ValidationError("Document has media without associated asset. Please ingest it or drop it.");
+                }
+            });
+        });
+    }
+
+    render() {
```

Comment:
Same thing I mentioned to @ptomato but this needs to be a real template in a separate file.